sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -0,0 +1,524 @@
+            return encrypted;
+          };
+        }
+      }
+    }
+
+    // Default to the service
+    let result = await this.client.wrapKey(this.vaultBaseUrl, this.name, this.version, algorithm, key, options);
+    return result.result!;
+  }
+
+  /**
+   * Unwraps the given wrapped key using the specified cryptography algorithm
+   * @param encryptedKey The encrypted key to unwrap
+   * @param algorithm The decryption algorithm to use to unwrap the key
+   * @param options Additional options
+   */
+  public async unwrapKey(
+    encryptedKey: Uint8Array,
+    algorithm: JsonWebKeyEncryptionAlgorithm,
+    options?: RequestOptions
+  ): Promise<Uint8Array> {
+    let result = await this.client.unwrapKey(this.vaultBaseUrl, this.name, this.version, algorithm, encryptedKey, options);
+    return result.result!;
+  }
+
+  /**
+   * Cryptographically sign the digest of a message
+   * @param digest The digest of the data to sign
+   * @param algorithm The signing algorithm to use
+   * @param options Additional options
+   */
+  public async sign(
+    digest: Uint8Array,
+    algorithm: JsonWebKeySignatureAlgorithm,
+    options?: RequestOptions
+  ): Promise<Uint8Array> {
+    let result = await this.client.sign(this.vaultBaseUrl, this.name, this.version, algorithm, digest, options);
+    return result.result!;
+  }
+
+  /**
+   * Verify the signed message digest
+   * @param digest The digest to verify
+   * @param signature The signature to verify the digest against
+   * @param algorithm The signing algorithm to use to verify with
+   * @param options Additional options
+   */
+  public async verify(
+    digest: Uint8Array,
+    signature: Uint8Array,
+    algorithm: JsonWebKeySignatureAlgorithm,
+    options?: RequestOptions
+  ): Promise<boolean> {
+    const response = await this.client.verify(this.vaultBaseUrl, this.name, this.version, algorithm, digest, signature, options);
+    return response.value ? response.value : false;
+  }
+
+  /**
+   * Cryptographically sign a block of data
+   * @param data The data to sign
+   * @param algorithm The signing algorithm to use
+   * @param options Additional options
+   */
+  public async signData(
+    data: Uint8Array,
+    algorithm: JsonWebKeySignatureAlgorithm,
+    options?: RequestOptions
+  ): Promise<Uint8Array> {
+    let digest;
+    switch (algorithm) {
+      case ("ES256"):
+      case ("ES256K"):
+      case ("PS256"):
+      case ("RS256"): {
+        digest = await CryptographyClient.createHash("sha256", data);
+      } break;
+      case ("ES384"):
+      case ("PS384"):
+      case ("RS384"): {
+        digest = await CryptographyClient.createHash("sha384", data);
+      } break;
+      case ("ES512"):
+      case ("PS512"):
+      case ("RS512"): {
+        digest = await CryptographyClient.createHash("sha512", data);
+      } break;
+      default: {
+        throw new Error("Unsupported verify algorithm");
```

Comment:
>throw new Error("Unsupported verify algorithm"); [](start = 8, length = 48)

Should be "Unsupported signature algorithm".  Also should this have a more specific error type?

Reply:
Will update. We generally just throw new Error.